stats/Record.md

```diff
@@ -68,5 +70,14 @@ MeasurementMap measurementMap = new MeasurementMap();
 measurementMap.put(RPC_LATENCY, 10.3);
 measurementMap.put(RPC_BYTES_SENT, 124);
 measurementMap.record();  // reads context from thread-local.
+
+// Another example on recording against sampled SpanContext.
+SpanContext spanContext = tracer.getCurrentSpan().getContext();
+if (spanContext.getTraceOptions().isSampled()) {
+  Map<String, String> attachments = new HashMap<>();
+  // Client code needs to take care of encoding.
+  attachments.put("TraceId", encode(spanContext.getTraceId));
+  attachments.put("SpanId", encode(spanContext.getSpanId));
```

Comment:
I think that's an implementation detail of the `encode()` method. Client code should decide whether or not they want the lowercase base16 encoding of TraceId/SpanId (there may also be clients that need some other encoding format).

Reply:
There needs to be agreement between the gRPC code and the Stackdriver exporter. Maybe we should make it a `Map<String, Object>` instead and pass the whole `spanContext`?

Reply:
This makes sense to me, by doing that client code doesn't need to worry about encoding too. Should we change the type of [exemplar.attachment](https://github.com/census-instrumentation/opencensus-proto/pull/62/files#diff-cbf22e3bb90f48f62fe55b894a9ee40cR236) to map<string, Any> to match this?

Reply:
Sorry to go in circles on this but we added the Map because we didn't want an explicit dependency from the stats API to the trace API. Maybe we should have instead just made it `Object`/`interface{}` and let the stats API just pass it through? To put it another way, what purpose is the key in the map serving?

That way only gRPC and the exporters need to know about the actual type. The stats API can just treat it as an opaque pointer/object and pass it along. Then `exemplar.attachment` in the proto can go back to being just `Any`.

Reply:
> To put it another way, what purpose is the key in the map serving?

FWIW we want a map here because there may be different types of Exemplar attachments for a single histogram bucket. For example, we can have both a set of dropped labels and a SpanContext associated with a histogram bucket. In that case, map keys are needed to differentiate attachments of different types (like a JSON object).

> That way only gRPC and the exporters need to know about the actual type. The stats API can just treat it as an opaque pointer/object and pass it along.

I think we can achieve the same goal with Map<String, Object>. The stats API doesn't need to know the actual type of the map value and can just pass it along.

Reply:
I think Object is too generic, because exporters won't be able to pass it along without knowing what it represents, since it's not possible to serialize unknown Objects.  Using String values gives exporters the flexibility to either look for specific keys that they understand, or export the whole map.